dotCMS/src/main/java/com/dotcms/rest/api/v1/template/TemplateResource.java

```diff
@@ -0,0 +1,752 @@
+    public final Response list(@Context final HttpServletRequest httpRequest,
+                                        @Context final HttpServletResponse httpResponse,
+                                        @QueryParam(PaginationUtil.FILTER)   final String filter,
+                                        @QueryParam(PaginationUtil.PAGE)     final int page,
+                                        @QueryParam(PaginationUtil.PER_PAGE) final int perPage,
+                                        @DefaultValue("title") @QueryParam(PaginationUtil.ORDER_BY) final String orderBy,
+                                        @DefaultValue("ASC") @QueryParam(PaginationUtil.DIRECTION)  final String direction,
+                                        @QueryParam(ContainerPaginator.HOST_PARAMETER_ID)           final String hostId) {
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requestAndResponse(httpRequest, httpResponse).rejectWhenNoUser(true).init();
+        final User user = initData.getUser();
+        final Optional<String> checkedHostId = Optional.ofNullable(Try.of(()-> APILocator.getHostAPI()
+                .find(hostId, user, false).getIdentifier()).getOrNull());
+
+        Logger.debug(this, ()-> "Getting the List of templates");
+
+        final Map<String, Object> extraParams = Maps.newHashMap();
+        checkedHostId.ifPresent(checkedHostIdentifier -> extraParams.put(ContainerPaginator.HOST_PARAMETER_ID, checkedHostIdentifier));
+        return this.paginationUtil.getPage(httpRequest, user, filter, page, perPage, orderBy, OrderDirection.valueOf(direction),
+                extraParams);
+    }
+
+    /**
+     * Return a {@link com.dotmarketing.portlets.templates.model.Template} based on the inode
+     *
+     * @return Response
+     */
+    @GET
+    @Path("/{templateInode}")
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response getByInode(@Context final HttpServletRequest  httpRequest,
+                               @Context final HttpServletResponse httpResponse,
+                               @PathParam("templateInode") final String templateInode) throws DotSecurityException, DotDataException {
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requestAndResponse(httpRequest, httpResponse).rejectWhenNoUser(true).init();
+        final User user     = initData.getUser();
+        final PageMode mode = PageMode.get(httpRequest);
+        Logger.debug(this, ()-> "Getting the template by inode: " + templateInode);
+
+        final Template template = this.templateAPI.find(templateInode, user, mode.respectAnonPerms);
+
+        if (null == template || UtilMethods.isNotSet(template.getIdentifier())) {
+
+            throw new DoesNotExistException("The template inode: " + templateInode + " does not exists");
```

Comment:
Codacy found an issue: [The String literal " does not exists" appears 4 times in this file; the first occurrence is on line 217](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6042990)